### PR TITLE
CLN: clean-up show_versions and consistently use null for json output

### PR DESCRIPTION
**Pull request by simonjayhawkins**
note changes to LC_ALL and LOCALE, null is already used for dependencies on master, see blosc and feather.

master
```
{
  "system": {
    "commit": "a7ecced88a42c426bf61016c0131cab023c0cdff",
    "python": "3.7.5.final.0",
    "python-bits": 64,
    "OS": "Windows",
    "OS-release": "10",
    "machine": "AMD64",
    "processor": "Intel64 Family 6 Model 58 Stepping 9, GenuineIntel",
    "byteorder": "little",
    "LC_ALL": "None",
    "LANG": "en_GB.UTF-8",
    "LOCALE": "None.None"
  },
  "dependencies": {
    "pandas": "1.1.0.dev0+497.ga7ecced88",
    "numpy": "1.17.2",
    "pytz": "2019.3",
    "dateutil": "2.8.0",
    "pip": "19.3.1",
    "setuptools": "41.6.0.post20191030",
    "Cython": "0.29.13",
    "pytest": "5.2.2",
    "hypothesis": "4.36.2",
    "sphinx": "2.2.1",
    "blosc": null,
    "feather": null,
...
```

this pr
```
{
  "system": {
    "commit": "2ac9d30f302e59de035cbea89188d5421212b000",
    "python": "3.7.5.final.0",
    "python-bits": 64,
    "OS": "Windows",
    "OS-release": "10",
    "Version": "10.0.18362",
    "machine": "AMD64",
    "processor": "Intel64 Family 6 Model 58 Stepping 9, GenuineIntel",
    "byteorder": "little",
    "LC_ALL": null,
    "LANG": "en_GB.UTF-8",
    "LOCALE": {
      "language-code": null,
      "encoding": null
    }
  },
  "dependencies": {
    "pandas": "1.1.0.dev0+500.g2ac9d30f3",
    "numpy": "1.17.2",
    "pytz": "2019.3",
    "dateutil": "2.8.0",
    "pip": "19.3.1",
    "setuptools": "41.6.0.post20191030",
    "Cython": "0.29.13",
    "pytest": "5.2.2",
    "hypothesis": "4.36.2",
    "sphinx": "2.2.1",
    "blosc": null,
...
```

console output is unchanged
```
INSTALLED VERSIONS
------------------
commit           : 2ac9d30f302e59de035cbea89188d5421212b000
python           : 3.7.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : None.None

pandas           : 1.1.0.dev0+500.g2ac9d30f3
numpy            : 1.17.2
pytz             : 2019.3
dateutil         : 2.8.0
pip              : 19.3.1
setuptools       : 41.6.0.post20191030
Cython           : 0.29.13
pytest           : 5.2.2
hypothesis       : 4.36.2
sphinx           : 2.2.1
blosc            : None
feather          : None
...
```

### CODE REVIEW

File: pandas/_typing.py [hunk 1]
```diff
@@ -64,7 +64,7 @@
 Label = Optional[Hashable]
 Level = Union[Label, int]
 Ordered = Optional[bool]
-JSONSerializable = Union[PythonScalar, List, Dict]
+JSONSerializable = Optional[Union[PythonScalar, List, Dict]]
```
Comment (simonjayhawkins) [inline pandas/_typing.py]: None is serializable

**Comment by simonjayhawkins** [comment]
hopefully the test added in #32041 will resolve the patch coverage here.

**Comment by jbrockmendel** [comment]
needs rebase

**Comment by jbrockmendel** [comment]
LGTM

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**